skimage/morphology/greyreconstruct.py

```diff
@@ -139,15 +144,20 @@ def reconstruction(seed, mask, method='dilation', selem=None, offset=None):
         if not all([d % 2 == 1 for d in selem.shape]):
             raise ValueError("Footprint dimensions must all be odd")
         offset = np.array([d // 2 for d in selem.shape])
+    else:
+        if offset.ndim != selem.ndim:
+            raise ValueError("offset and selem ndims must be identical")
+        if not all([(0 <= o < d) for o, d in zip(offset, selem.shape)]):
+            raise ValueError("Offset must be included inside selem")
+
     # Cross out the center of the selem
     selem[tuple(slice(d, d + 1) for d in offset)] = False
 
     # Make padding for edges of reconstructed image so we can ignore boundaries
-    padding = (np.array(selem.shape) / 2).astype(int)
     dims = np.zeros(seed.ndim + 1, dtype=int)
-    dims[1:] = np.array(seed.shape) + 2 * padding
+    dims[1:] = np.array(seed.shape) + (np.array(selem.shape) - 1).astype(int)
```

Comment:
Is it still necessary to cast the type? (here and below)

Reply:
No of course you're right , the fix is in https://github.com/scikit-image/scikit-image/pull/3591/commits/2d0f9fa2bdf1844a63fa92e709a4407d966790a5